pennylane/ops/op_math/sprod.py

```diff
@@ -114,6 +114,16 @@ def __init__(self, scalar: Union[int, float, complex], base: Operator, do_queue=
         self._check_scalar_is_valid()
         super().__init__(base=base, do_queue=do_queue, id=id)
 
+        try:
```

Comment:
Maybe instead using `if getattr(self.base, "_pauli_rep", None) is not None`?

That might bypass having a try-except block.

Reply:
Good Idea! adding it in now!